tests/unit/test_schedule.cpp

```diff
@@ -0,0 +1,280 @@
+#include <algorithm>
+#include <random>
+#include <stdexcept>
+#include <vector>
+
+#include <common_types.hpp>
+#include <schedule.hpp>
+#include <util/partition.hpp>
+#include <util/rangeutil.hpp>
+
+#include "common.hpp"
+#include "stats.hpp"
+
+using namespace nest::mc;
+using namespace testing;
+
+// Pull events from n non-contiguous subintervals of [t0, t1)
+// and check for monotonicity and boundedness.
+
+void run_invariant_checks(schedule S, time_type t0, time_type t1, unsigned n, int seed=0) {
+    if (!n) return;
+
+    std::minstd_rand R(seed);
+    std::uniform_real_distribution<time_type> U(t0, t1);
+
+    std::vector<time_type> divisions = {t0, t1};
+    std::generate_n(std::back_inserter(divisions), 2*(n-1), [&] { return U(R); });
+    util::sort(divisions);
```

Comment:
What if there are repeated values in `divisions`? Won't this violate the condition that each interval is half open?

Reply:
That should be fine: it's just then the empty half-open interval _[t, t)_ = Ø.